packages/shared_preferences/shared_preferences/lib/src/shared_preferences_async.dart

```diff
@@ -184,6 +184,20 @@ class SharedPreferencesWithCache {
             SharedPreferencesAsync(options: sharedPreferencesOptions),
         _cache = cache ?? <String, Object?>{};
 
+  /// Creates a new instance with the given options
```

Comment:
```suggestion
  /// Creates a new instance with the given options, but does not populate the
  /// cache from the platform.
  ///
  /// This constructor is synchronous and does not load data from disk, which
  /// can be useful for dependency injection frameworks that do not support
  /// asynchronous object creation. The cache must be populated by calling
  /// [reloadCache] before it can be used.
```